File: spring-web-java/src/main/java/spring/web/java/infrastructure/persistence/adapter/NotificationRepositoryAdapter.java
package spring.web.java.infrastructure.persistence.adapter;

import org.springframework.stereotype.Component;

import lombok.RequiredArgsConstructor;
import spring.web.java.domain.model.entity.Notification;
import spring.web.java.domain.repository.NotificationRepository;
import spring.web.java.infrastructure.persistence.NotificationJpaRepository;

@Component
@RequiredArgsConstructor
public class NotificationRepositoryAdapter implements NotificationRepository {

	private final NotificationJpaRepository jpaRepository;

	@Override
	public Notification save(Notification notification) {
		return jpaRepository.save(notification);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/persistence/adapter/NotificationRepositoryAdapter.kt
package spring.web.kotlin.infrastructure.persistence.adapter

import org.springframework.stereotype.Component
import spring.web.kotlin.domain.model.entity.Notification
import spring.web.kotlin.domain.repository.NotificationRepository
import spring.web.kotlin.infrastructure.persistence.NotificationJpaRepository

@Component
class NotificationRepositoryAdapter(
    private val jpaRepository: NotificationJpaRepository
) : NotificationRepository {
    override fun save(notification: Notification) = jpaRepository.save(notification)
}